packages/react-instantsearch/src/components/Autocomplete.tsx

```diff
@@ -17,14 +38,38 @@ export type AutocompleteProps = {
 };
 
 export function EXPERIMENTAL_Autocomplete({ indices }: AutocompleteProps) {
+  const [isOpen, setIsOpen] = useState(false);
+
   return (
     <Index EXPERIMENTAL_isolated>
-      <SearchBox />
-      {indices.map((index) => (
-        <Index key={index.indexName} indexName={index.indexName}>
-          <Hits hitComponent={({ hit }) => <index.itemComponent {...hit} />} />
-        </Index>
-      ))}
+      <Autocomplete isOpen={isOpen}>
+        <SearchBox onFocus={() => setIsOpen(true)} />
+        <AutocompletePanel isOpen={isOpen}>
+          {indices.map((index) => (
+            <Index key={index.indexName} indexName={index.indexName}>
+              <AutocompleteIndexComponent itemComponent={index.itemComponent} />
+            </Index>
+          ))}
+        </AutocompletePanel>
+      </Autocomplete>
     </Index>
   );
 }
+
+type AutocompleteIndexProps = {
+  itemComponent: IndexConfig['itemComponent'];
+};
+
+function AutocompleteIndexComponent({
+  itemComponent: ItemComponent,
+}: AutocompleteIndexProps) {
+  const { items } = useHits();
+
+  return (
+    <AutocompleteIndex
+      // @ts-expect-error - there seems to be problems with React.ComponentType and this, but it's actually correct
```

Comment:
should we cast it instead?

Reply:
Doesn't work, but it's sensibly the same as adding `ts-expect-error` (except if the line had more statements than this)